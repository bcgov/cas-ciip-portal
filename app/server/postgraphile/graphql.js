const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
