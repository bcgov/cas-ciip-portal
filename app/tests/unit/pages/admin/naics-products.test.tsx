import React from "react";
import NaicsProducts from "pages/admin/naics-products";
import { shallow } from "enzyme";
import { naicsProductsAssociationsQueryResponse } from "__generated__/naicsProductsAssociationsQuery.graphql";

describe("the product naics page", () => {
  const query: naicsProductsAssociationsQueryResponse["query"] = {
    " $fragmentRefs": { ProductNaicsCodeAssociation_query: true },
    session: {
      " $fragmentRefs": { defaultLayout_session: true },
    },
  };

  it("matches the last accepted Snapshot", () => {
    const wrapper = shallow(<NaicsProducts query={query} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("passes a query to the association component", () => {
    const wrapper = shallow(<NaicsProducts query={query} />);
    expect(
      wrapper.find("Relay(ProductNaicsCodeAssociationContainer)").prop("query")
    ).toBe(query);
  });
});
