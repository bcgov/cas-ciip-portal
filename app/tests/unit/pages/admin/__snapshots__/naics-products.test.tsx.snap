// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the product naics page matches the last accepted Snapshot 1`] = `
<Relay(DefaultLayout)
  session={
    Object {
      " $fragmentRefs": Object {
        "defaultLayout_session": true,
      },
    }
  }
  title="Allowable Products by Industry (NAICS)"
>
  <Relay(ProductNaicsCodeAssociationContainer)
    query={
      Object {
        " $fragmentRefs": Object {
          "ProductNaicsCodeAssociation_query": true,
        },
        "session": Object {
          " $fragmentRefs": Object {
            "defaultLayout_session": true,
          },
        },
      }
    }
  />
</Relay(DefaultLayout)>
`;
